scripts/test/runtests.sh

```diff
@@ -81,4 +85,7 @@ done
 popd
 set -e
 
+#resetting the path so that we can be good neighbors to other scripts running after this.
+export PATH=$StartPath
```

Comment:
All right, I will remove it, though I will only add it back once your refactoring goes in.